code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -138,3 +139,93 @@
 
 /obj/item/gun/energy/laser/redtag/hitscan
 	ammo_type = list(/obj/item/ammo_casing/energy/laser/redtag/hitscan)
+
+
+//////////////EXTERNAL LENS////////////////////////////
+
+/obj/item/external_lens
+	name = "external lens"
+	icon_state = "external"
+	desc = "external lens."
+	icon = 'icons/obj/guns/energy.dmi'
+	var/stored_ammo_type = /obj/item/ammo_casing/energy/laser
+	var/overlay = "laser"
+
+/obj/item/external_lens/Initialize()
+	. = ..()
+	add_overlay(overlay)
+
+/obj/item/external_lens/afterattack(atom/movable/AM, mob/user, flag)
+	. = ..()
+	if(istype(AM, /obj/item/gun/energy/laser))
+		var/obj/item/gun/energy/laser/L = AM
+		if(!L.GetComponent(/datum/component/extralasers))

```

Comment:
You do not need to GetComponent here, if you're unsure how to use components contact me with questions somewhere

Reply:
what do i need then

Reply:
I said earlier, you can make use of how components inherit to make all this get handled in the component

Reply:
do i have to create the component on initialize inside the lens and then move it into the gun?

Reply:
No, components have special handling for what happens if you apply a component to something that already has that kind of component. You can make use of that to handle what happens when you try to apply a lens on top of another lens